src/Catalyst.Core.Modules.Kvm.Tests/IntegrationTests/CatalystVirtualMachineTests.cs

```diff
@@ -188,6 +194,70 @@ public void Can_Invoke_Chain_Id()
             trace.Entries.Last().Stack.Last().Should().Be(VirtualMachine.BytesOne32.ToHexString());
         }
 
+        [Fact]
+        [Trait(Traits.TestType, Traits.IntegrationTest)]
+        public void Blake_precompile()
+        {
+            Address blakeAddress = Address.FromNumber(1 + KatVirtualMachine.CatalystPrecompilesAddressingSpace);
+            string addressCode = blakeAddress.Bytes.ToHexString(false);
+            var code = Bytes.FromHexString("0x602060006080600073" + addressCode + "45fa00");
+            var txTracer = RunVirtualMachine(code);
+            var serializer = new EthereumJsonSerializer();
+            var trace = txTracer.BuildResult();
+            _testOutputHelper.WriteLine(serializer.Serialize(trace, true));
+            trace.Entries.Last().Stack.First().Should().Be("0000000000000000000000000000000000000000000000000000000000000001");
+            trace.Entries.Last().Memory.First().Should().Be("378d0caaaa3855f1b38693c1d6ef004fd118691c95c959d4efa950d6d6fcf7c1");
+        }
+        
+        [Fact]
+        [Trait(Traits.TestType, Traits.IntegrationTest)]
+        public void Ed25519_verify_precompile()
+        {
+            HashProvider hashProvider = new HashProvider(HashingAlgorithm.GetAlgorithmMetadata("blake2b-256"));
+            
+            FfiWrapper cryptoContext = new FfiWrapper();
+            var privateKey = cryptoContext.GeneratePrivateKey();
+            
+            var data = Encoding.UTF8.GetBytes("Testing testing 1 2 3");
+            var message = hashProvider.ComputeMultiHash(data).Digest;
+            
+            var signingContext = Encoding.UTF8.GetBytes("Testing testing 1 2 3 context");
+            var context = hashProvider.ComputeMultiHash(signingContext).Digest;
+            
+            var signature = cryptoContext.Sign(privateKey, message, context);
+            var signatureBytes = signature.SignatureBytes;
+            
+            var publicKeyBytes = privateKey.GetPublicKey().Bytes;
```

Comment:
it's not important here but just to point out ```privateKey.GetPublicKey()``` invokes the ffi again, whereas the public key bytes are already available from ```signature.PublicKeyBytes```.

Reply:
yes, this needs to get fixed, let me fix it on Monday

Reply:
we want to minimize the calls -this is a proof of concept - it works and it is fine to merge but would love to hear the details on when you would use it and how -then we can adjust it a bit

Reply:
it was just a test logic, I keep it since I made some changes that split the valid / invalid key code location